frontend/src/scenes/events/eventsTableLogic.ts

```diff
@@ -33,64 +38,109 @@ const formatEvents = (events, newEvents, apiUrl) => {
+            props.fixedFilters ? JSON.stringify(props.fixedFilters) : 'all',
+            props.apiUrl || 'events',
+            props.live ? 'live' : '',
+            props.key,
+        ].join('-'),
+
+    actions: {
+        setProperties: (properties: Record<string, unknown>[]) => {
+            // there seem to be multiple representations of "empty" properties
+            // the page does not work with some of those representations
+            // this action normalises them
+            if (Array.isArray(properties)) {
+                if (properties.length === 0) {
+                    return { properties: [{}] }
+                } else {
+                    return { properties }
+                }
+            } else {
+                return { properties: [properties] }
             }
-            return { properties }
         },
-        setColumnConfig: (columnConfig) => ({ columnConfig }),
-        setColumnConfigSaving: (saving) => ({ saving }), // Type: boolean
+        setColumnConfig: (columnConfig: string[]) => ({ columnConfig }),
+        setColumnConfigSaving: (saving: boolean) => ({ saving }),
         fetchEvents: (nextParams = null) => ({ nextParams }),
-        fetchEventsSuccess: (events, hasNext = false, isNext = false) => ({ events, hasNext, isNext }),
+        fetchEventsSuccess: (x: OnFetchEventsSuccess) => x,
         fetchNextEvents: true,
-        fetchOrPollFailure: (error) => ({ error }),
+        fetchOrPollFailure: (error: ApiError) => ({ error }),
         flipSort: true,
         pollEvents: true,
-        pollEventsSuccess: (events) => ({ events }),
-        prependNewEvents: (events) => ({ events }),
-        setSelectedEvent: (selectedEvent) => ({ selectedEvent }),
-        setPollTimeout: (pollTimeout) => ({ pollTimeout }),
+        pollEventsSuccess: (events: EventType[]) => ({ events }),
+        prependNewEvents: (events: EventType[]) => ({ events }),
+        setSelectedEvent: (selectedEvent: EventType) => ({ selectedEvent }),
+        setPollTimeout: (pollTimeout: number) => ({ pollTimeout }),
         setDelayedLoading: true,
-        setEventFilter: (event) => ({ event }),
-        toggleAutomaticLoad: (automaticLoadEnabled) => ({ automaticLoadEnabled }),
-    }),
+        setEventFilter: (event: string) => ({ event }),
+        toggleAutomaticLoad: (automaticLoadEnabled: boolean) => ({ automaticLoadEnabled }),
+        noop: (s: string): string => s,
+    },
 
-    reducers: () => ({
+    reducers: {
         // save the pathname that was used when this logic was mounted
         // we use it to NOT update the filters when the user moves away from this path, yet the scene is still active
-        initialPathname: [(state) => router.selectors.location(state).pathname, { noop: (a) => a }],
+        initialPathname: [
+            (state: string) => router.selectors.location(state).pathname,
+            { noop: (_: string, s: string) => s },
+        ],
         properties: [
             [],
```

Comment:
This should be typed as `[] as Something[]`

Reply:
I think I've understood you and resolved this....